packages/databricks-vscode/src/workspace-fs/WorkspaceFsAccessVerifier.ts

```diff
@@ -0,0 +1,106 @@
+import {Cluster} from "@databricks/databricks-sdk";
+import {commands, Disposable, window} from "vscode";
+import {ConnectionManager} from "../configuration/ConnectionManager";
+import {CodeSynchronizer} from "../sync";
+import {workspaceConfigs} from "../vscode-objs/WorkspaceConfigs";
+
+async function switchToRepos() {
+    await workspaceConfigs.setSyncDestinationType("repo");
+    commands.executeCommand("workbench.action.reloadWindow");
+}
+
+export class WorkspaceFsAccessVerifier implements Disposable {
+    private disposables: Disposable[] = [];
+    private currentCluster: Cluster | undefined;
+
+    constructor(
+        private _connectionManager: ConnectionManager,
+        private _sync: CodeSynchronizer
+    ) {
+        this.disposables.push(
+            this._connectionManager.onDidChangeCluster(async (cluster) => {
+                if (this.currentCluster?.name === cluster?.name) {
+                    return;
+                }
+                this.currentCluster = cluster;
+                await this.verifyCluster(cluster);
+            }),
+            this._sync.onDidChangeState(async (state) => {
+                if (
+                    workspaceConfigs.syncDestinationType === "repo" &&
+                    state === "FILES_IN_REPOS_DISABLED"
+                ) {
+                    await window.showErrorMessage(
+                        "Sync failed. Files in Repos is disabled for the current workspace. Please contact your system admin to enable it for your workspace."
+                    );
+                } else if (
+                    workspaceConfigs.syncDestinationType === "workspace" &&
+                    state === "FILES_IN_WORKSPACE_DISABLED"
+                ) {
+                    const selection = await window.showErrorMessage(
+                        "Sync failed. Files in Workspace is disabled for the current workspace.",
+                        "Switch to Repos",
+                        "Ignore"
+                    );
+
+                    if (selection === "Switch to Repos") {
+                        switchToRepos();
+                    }
+                }
+            })
+        );
+    }
+
+    async verifyCluster(cluster?: Cluster) {
+        if (
+            workspaceConfigs.syncDestinationType === "repo" ||
+            cluster === undefined
+        ) {
+            return;
+        }
+        const dbrVersionParts = cluster.dbrVersion;
+        if (
+            dbrVersionParts[0] < 11 ||
+            (dbrVersionParts[0] === 11 &&
+                dbrVersionParts[1] !== "x" &&
+                dbrVersionParts[1] < 2)
+        ) {
+            const message =
+                "Files in workspace is not supported on clusters with DBR < 11.2.";
```

Comment:
I wonder if users understand this message without context. "Why does this matter?"

We might want to elaborate a bit more.